### ISSUE TITLE
`array_slice('duckDB', 5, NULL)` gives `NULL`, not `"DB"`

### ISSUE THREAD

**Issue by NickCrews** [duckdb/duckdb#11431]
### What happens?

```python
import duckdb

c = duckdb.connect()
print(c.sql("SELECT array_slice('DuckDB', 5, NULL);"))
print(c.sql("SELECT array_slice('DuckDB', NULL, 3);"))
print(c.sql("SELECT array_slice('DuckDB', NULL, NULL);"))
```

All three of these show NULL. Based on https://duckdb.org/docs/sql/functions/char.html#array_slicelist-begin-end
these should give "DB", "Duc", and "DuckDB", respectively, consistent with how python's string slicing works.

Ran into this when trying to implement https://github.com/ibis-project/ibis/issues/8796

### To Reproduce

```sql
SELECT array_slice('DuckDB', 5, NULL);
```

### OS:

macos m1

### DuckDB Version:

duckdb-0.10.2.dev311

### DuckDB Client:

python

### Full Name:

Nick Crews

### Affiliation:

Ship Creek Group

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by taniabogatsch** [comment]
Hi @NickCrews. We changed our slicing behavior to be compatible with postgres after 0.8.0. It seems our documentation is describing the behavior of the previous version. I've filed an issue to update the documentation here: https://github.com/duckdb/duckdb-web/issues/2655. I'll close this issue as intended behavior.

The original issue prompting the changes: https://github.com/duckdb/duckdb/issues/3176.
